senders/proxy.go

```diff
@@ -73,11 +73,9 @@ func (sender *proxySender) SendMetric(name string, value float64, ts int64, sour
 		return fmt.Errorf("proxy metrics port not provided, cannot send metric data")
 	}
 
-	if !sender.metricHandler.Connected() {
-		err := sender.metricHandler.Connect()
-		if err != nil {
-			return err
-		}
+	err := sender.metricHandler.Connect()
```

Comment:
`Connected()` uses a read lock, whereas `Connect()` uses a write lock. It'd better to retain the previous approach for concurrent operations.